### ISSUE TITLE
Vagrant & Bazel Experiences

### ISSUE THREAD

**Issue by jhh** [frc1678/vagrant-box#1]
I just started mentoring for 2767 in Kalamazoo, MI and was looking to move our build system over to vagrant and Bazel. I was wondering how it was working for you?

**Comment by donald-pinckney** [comment]
Hi Jeff,

Last year we used Bazel without Vagrant. We mostly had students running some Linux distro on their personal laptops, and a few team laptops running Debian, all setup with Bazel. I would say that our experience with Bazel has been positive; every once in a while you run into some frustrating things with Bazel, but hey, no build system is perfect.    I feel like writing our code on Linux and using Bazel to compile lead to faster development, as compared to the standard FRC eclipse environment.    Using Linux gives you much more convenient access to other tools such as gnuplot, which we used a lot.

This year we don't plan on moving wholesale to Vagrant.    We are introducing Vagrant for 2 reasons:
1. For teaching programming in fall, we are using Swift, which currently needs either Mac OS X or Linux, and can be somewhat finicky to setup.    This Vagrant box provides a standard Linux environment for Swift, that any new student should be able to run on any machine, even if they have Windows.
1. This also provides a standard Linux environment with Bazel setup properly, and other tools such as numpy, etc. setup, so that programmers without personal Linux machines can still easily build and run our robot code.    We will still let programmers use their own setup rather than Vagrant, but if they have build errors it's up to them to fix their machine configuration, or move to using Vagrant.

If you have more technical questions about Bazel, just let me know and I can put you in contact with Kyle Stachowitz, a student and current Bazel wizard.

For Vagrant, feel free to use this Vagrant box, or fork it and tweak it, if you don't want Swift for example.

Donald Pinckney
